packages/core/src/routes/application.ts

```diff
@@ -46,19 +47,32 @@ export default function applicationRoutes<T extends AuthedRouter>(
   router.get(
     '/applications',
     koaPagination({ isOptional: true }),
-    koaGuard({ response: z.array(Applications.guard), status: 200 }),
+    koaGuard({
+      response: z.array(Applications.guard),
+      status: 200,
+    }),
     async (ctx, next) => {
       const { limit, offset, disabled: paginationDisabled } = ctx.pagination;
+      const { searchParams } = ctx.URL;
+
+      // TODO: koaGuard() can not handle array in query for now (LOG-7080).
+      const result = z.nativeEnum(ApplicationType).array().safeParse(searchParams.getAll('type'));
+      if (!result.success) {
+        throw new RequestError({ code: 'guard.invalid_input', type: 'query' }, result.error);
+      }
```

Comment:
? Should do the the type casting at the query level.

Reply:
Updated, reuse search for type filtering.